clients/drcachesim/tests/invariant_checker_test.cpp

```diff
@@ -328,6 +331,73 @@ check_kernel_xfer()
         if (!run_checker(memrefs, false))
             return false;
     }
+    // Multiple nested signals without any intervening instr between the nested
+    // signals.
+    {
+        std::vector<memref_t> memrefs = {
+            gen_instr(1, 1),
+            gen_marker(1, TRACE_MARKER_TYPE_KERNEL_EVENT, 2),
+            gen_instr(1, 101),
+            // First nested signal.
+            gen_marker(1, TRACE_MARKER_TYPE_KERNEL_EVENT, 102),
+            gen_instr(1, 201),
+            // XXX: This marker value is actually not guaranteed, yet the checker
+            // requires it and the view tool prints it.
+            gen_marker(1, TRACE_MARKER_TYPE_KERNEL_XFER, 202),
+            // No intervening instr here. Should use instr at pc = 101 for
+            // pre-signal instr check on return.
+            // Second nested signal.
+            gen_marker(1, TRACE_MARKER_TYPE_KERNEL_EVENT, 102),
+            gen_instr(1, 201),
+            // XXX: This marker value is actually not guaranteed, yet the checker
+            // requires it and the view tool prints it.
+            gen_marker(1, TRACE_MARKER_TYPE_KERNEL_XFER, 202),
+            gen_instr(1, 102),
+            gen_marker(1, TRACE_MARKER_TYPE_KERNEL_XFER, 103),
+            gen_instr(1, 2),
+        };
+        if (!run_checker(memrefs, false))
+            return false;
+    }
+    // Trace starts in a signal.
+    {
+        std::vector<memref_t> memrefs = {
+            // Already inside the first signal.
+            gen_instr(1, 11),
+            // XXX: This marker value is actually not guaranteed, yet the checker
```

Comment:
What does "not guaranteed" mean?  Why is this different from any other sigreturn/etc.?  This marker value is required to be correct and the checker should be checking it.

Ditto below.

Reply:
This comment is same as many other similar comments in the test file. I think it points to the fact that a sigreturn is not necessary when coming back from a signal.

Reply:
OK, so `s/This marker value/This marker/` and `s/guaranteed/guaranteed to be present/` -- though this is a particular sequence in a test so I would just remove this comment altogether.

Reply:
Removed.